File: monopoly/app/models/player.rb
# == Schema Information
#
# Table name: players
#
#  id         :integer          not null, primary key
#  user_id    :integer
#  game_id    :integer
#  name       :string
#  token      :string
#  cash       :integer
#  in_jail    :boolean
#  jail_rolls :integer
#  position   :integer
#  created_at :datetime         not null
#  updated_at :datetime         not null
#

class Player < ActiveRecord::Base
  belongs_to :game
  belongs_to :user
  has_many :owned_properties
  validates :name, presence: true
  validates :cash, presence: true
  validates_inclusion_of :in_jail, :in => [true, false]
  validates :position, presence: true
  after_initialize :init

  def init
    self.cash ||= 1500
    self.in_jail ||= false
    self.position ||= 0
  end
end

File: ios/Monopoly/Monopoly/Player.swift
//
//  Player.swift
//  Monopoly
//
//  Created by Tim Stephani on 7/18/15.
//  Copyright (c) 2015 Tim Stephani. All rights reserved.
//

class Player {
    var id : Int
    //var user_id : Int
    var name : String
    var cash : Int
    var in_jail : Bool
    var position : Int
    
    var user : User?
    
    init(id: Int, name: String, cash: Int, in_jail: Bool, position: Int) {
        self.id = id
        self.name = name
        self.cash = cash
        self.in_jail = in_jail
        self.position = position
    }
    
}
